internal/handler/site/middleware.go

```diff
@@ -1,16 +0,0 @@
-package site
-
-import (
-	"net/http"
-
-	"github.com/oursky/pageship/internal/site"
-)
-
-type Middleware func(*site.Descriptor, http.Handler) http.Handler
-
-func applyMiddleware(site *site.Descriptor, middlewares []Middleware, handler http.Handler) http.Handler {
```

Comment:
Why need move this file?

Reply:
because putting it in site would lead to a import loop (site/middleware.go -> middleware/xxx.go -> site)

Reply:
However `site/middleware.go` does not imports `site/middleware/*.go`?

Reply:
sorry, the root cause was because originally `site/middleware/middleware.go` imported `site/middleware.go`, and changing `site/middleware/middleware.go/func Default` into a closure meant `func Default -> func NewCacheContext (in middleware/cache.go) -> (site_)handler.go -> middleware`

Reply:
Both `func Default` and `func NewCacheContext` are within `handler/site/middleware` package, and `site_handler.go` is within `handler/site` package.

I don't see why `site_handler.go` need to import `handler/site/middleware` package? It only uses `Middleware` type and `ApplyMiddleware` function from the package, but originally both are within `handler/site` package.

Reply:
I see now there was a ContentCacheType declared in middleware/cache.go which lead to some problems. After separating the 2 middleware.go however, site/handler.go imports middleware because it needs middleware.Middleware (now in site/middleware), but middleware imports site because of middleware/middleware.go needs site.Middleware type. Even if i move around some definitions, i cannot fix the fact that the two middleware.go rely on each other,